tooling/internal/python/types/types.go

```diff
@@ -159,7 +170,26 @@ func writeUnionClass(w *formatting.IndentedWriter, className string, typeParamet
 }
 
 func writeNamedType(w *formatting.IndentedWriter, td *dsl.NamedType) {
-	fmt.Fprintf(w, "%s = %s\n", common.TypeIdentifierName(td.Name), common.TypeSyntax(td.Type, td.Namespace))
+	// Does this NamedType resolve to a RecordDefinition?
+	resolvesToRecord := false
+	dsl.Visit(td, func(self dsl.Visitor, node dsl.Node) {
```

Comment:
This doesn't work in all cases, for instance if there is a record within a vector.

Could you maybe call `GetUnderlyingType` and then see if it's a simpletype pointing to a RecordDefinition?

Reply:
Yep, you're correct. I fixed this and added a type to the test model to demonstrate your example (record within a vector).